firefox-ios/Client/Frontend/Autofill/LoginStorage.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Storage
+
+protocol LoginStorage {
+    func listLogins() async throws -> [EncryptedLogin]
+}
+
+extension RustLogins: LoginStorage {
+    func listLogins() async throws -> [EncryptedLogin] {
+        return try await withCheckedThrowingContinuation { continuation in
+            self.listLogins().upon { result in
+                switch result {
+                case .success(let logins):
+                    continuation.resume(returning: logins)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+}
```

Comment:
I think this should be part of the `RustLogins.swift` file instead of being under the `Autofill` folder. This will easily be missed if someone else works on the RustLogins. The protocol you added can be directly in that file.

Reply:
LoginStorage is part of our target because we need a way to create mocks for SwiftUI previews. The conformation cannot happen there since RustLogins has no idea what LoginStorage is.

Reply:
I'm not sure I understand. This protocol is just an async await enclosure on the Deferred method `public func listLogins() -> Deferred<Maybe<[EncryptedLogin]>>` no?

Reply:
`RustLogins.swift` is in the `Storage` framework. `LoginListViewModel` and `LoginAutofillView` are in our main target. I only need LoginStorage as a way to mock the logins for SwiftUI previews so I don't depend on Storage. If I move LoginStorage in the Storage framework then I will couple the views with a frameworks that does not need to know about any storage logic.